indico/modules/rb_new/client/js/components/TimelineItem.jsx

```diff
@@ -69,27 +78,54 @@ export default class TimelineItem extends React.Component {
         return (segmentStartDuration - (startHour * 60)) / (step * 60);
     };
 
-    renderOccurrence = (occurrence, additionalClasses = '') => {
-        const {start_dt: startDt, end_dt: endDt, reservation, bookable} = occurrence;
+    renderOccurrence = (occurrence, additionalClasses = '', type = '') => {
+        let segmentStartDt, segmentEndDt, popupContent;
+        const {start_dt: startDt, end_dt: endDt, start_time: startTime, end_time: endTime,
+               reservation, reason, bookable} = occurrence;
         const {startHour, endHour, step, onClick} = this.props;
-        const segmentStartDt = moment(startDt, 'YYYY-MM-DD HH:mm');
-        const segmentEndDt = moment(endDt, 'YYYY-MM-DD HH:mm');
+        if (type === 'blocking') {
+            segmentStartDt = moment(startHour, 'HH:mm');
+            segmentEndDt = moment(endHour, 'HH:mm');
+        } else if (type === 'unbookable-hours') {
+            segmentStartDt = moment(startTime, 'HH:mm');
+            segmentEndDt = moment(endTime, 'HH:mm');
+        } else {
+            segmentStartDt = moment(startDt, 'YYYY-MM-DD HH:mm');
+            segmentEndDt = moment(endDt, 'YYYY-MM-DD HH:mm');
+        }
         const blockWidth = 100 / (((endHour - startHour) / step) + 1);
         const segmentWidth = this.calculateWidth(segmentStartDt, segmentEndDt) * blockWidth;
         const segmentPosition = this.calculatePosition(segmentStartDt) * blockWidth;
-
-        const popupContent = (
-            <>
-                <div>
-                    {segmentStartDt.format('HH:mm')} - {segmentEndDt.format('HH:mm')}
+        if (type === 'blocking') {
+            popupContent = (
+                <div styleName="popup-center">{Translate.string('Room blocked: {reason}', {reason})}</div>
+            );
+        } else if (type === 'unbookable-periods') {
+            popupContent = (
+                <div styleName="popup-center">{Translate.string('Not possible to book in this period')}</div>
+            );
+        } else if (type === 'unbookable-hours') {
+            popupContent = (
+                <div styleName="popup-center">
+                    <div>{Translate.string('Not possible to book between:')}</div>
+                    <div>{segmentStartDt.format('HH:mm')} - {segmentEndDt.format('HH:mm')}</div>
                 </div>
-                <div>
-                    {reservation ? reservation.booking_reason : (bookable ? Translate.string('Click to book it') : '')}
+            );
+        } else {
+            popupContent = (
+                <div styleName="popup-center">
+                    <div>
+                        {segmentStartDt.format('HH:mm')} - {segmentEndDt.format('HH:mm')}
+                    </div>
+                    <div>
+                        {reservation ? reservation.booking_reason : (bookable ? Translate.string('Click to book it') : '')}
```

Comment:
i think nesting ternaries isn't pretty - i'd prather use a normal if..elseif..else, assign to a variable, and then use that here.